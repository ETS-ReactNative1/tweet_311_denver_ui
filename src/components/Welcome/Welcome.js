import React from 'react';
import { View, StyleSheet, Image, Text, Dimensions, TouchableOpacity } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';

const height = Dimensions.get('window').height;
const width = Dimensions.get('window').width;

export const Welcome = () => {

  return(
    <View style={styles.welcomeContainer}>
    <LinearGradient
          colors={['#4c669f', '#3b5998', '#192f6a']}
          style={styles.linearGradient}>
      <Image style={styles.img} source={require('../../../assets/images/denver.png')}></Image>
      <Text style={styles.welcomeText}>Welcome!</Text>
      <Text style={styles.welcomePara}>
        Thank you for using Tweet311Denver. We aim to build a safer, cleaner, 
        and more enjoyable commute experience for bikers, pedestrians, and 
        public transportation enthusiasts. Submit a claim below and contribute 
        to positive change!
      </Text>
      <TouchableOpacity style={styles.startButton}>
        <Text style={styles.startText}>Start Form</Text>
      </TouchableOpacity>
      </LinearGradient>
    </View>
  )
};

export default Welcome;

const styles = StyleSheet.create({
  welcomeContainer: {
    backgroundColor: '#fff',
    flexDirection: 'column',
    flex: 1,
    width: width * 1,
  },
  img: {
    height: 300,
    marginTop: height * .09,
    marginLeft: width * .05,
    width: 300,
  },
  linearGradient: {
    height: height * 1, 
    width: width * 1,
  },
  welcomeText: {
    color: '#FFFFFF',
    fontSize: 50,
    fontWeight: 'bold',
    marginLeft: width * .05,
    marginBottom: 15,
  },
  welcomePara: {
    color: '#FFFFFF',
    fontSize: 18,
    marginLeft: width * .05,
  },
  startText: {
    color: '#FFFFFF',
    fontSize: 20
  },
  startButton: {
    alignItems: 'center',
    backgroundColor: '#3976EA',
    borderRadius: 20,
    height: 40,
    justifyContent: 'center',
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: 50,
    width: width * .5,
  },
});
