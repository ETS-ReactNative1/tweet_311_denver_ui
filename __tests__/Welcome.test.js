import React from 'react';
import renderer from 'react-test-renderer';

import Welcome from '../src/components/Welcome/Welcome';

test('renders Success correctly', () => {
  const tree = renderer.create(<Welcome />).toJSON();
  expect(tree).toMatchSnapshot();
});
