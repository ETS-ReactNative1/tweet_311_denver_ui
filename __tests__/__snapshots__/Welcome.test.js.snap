// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Success correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "width": 750,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4283197087,
          4282079640,
          4279840618,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "height": 1334,
        "width": 750,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 300,
          "marginLeft": 22.5,
          "marginTop": 200.1,
          "width": 300,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 50,
          "fontWeight": "bold",
          "marginBottom": 15,
          "marginLeft": 30,
        }
      }
    >
      Welcome!
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
          "marginLeft": 30,
          "marginRight": 22.5,
        }
      }
    >
      Thank you for using 
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Tweet311Denver
      </Text>
      . We aim to build a safer, cleaner, and more enjoyable commute experience for bikers, pedestrians, and public transportation enthusiasts. Submit a claim below and contribute to positive change!
    </Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3976EA",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "opacity": 1,
          "width": 375,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Start Form
      </Text>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
